sos/plugins/__init__.py

```diff
@@ -537,7 +537,10 @@ def add_copy_spec(self, copyspecs, sizelimit=None, tailit=True):
             _file = None
 
             for _file in files:
-                current_size += os.stat(_file)[stat.ST_SIZE]
+                try:
+                    current_size += os.stat(_file)[stat.ST_SIZE]
+                except (OSError, FileNotFoundError):
+                    self._log_info("failed to stat '%s'" % _file)
```

Comment:
Yep, looks good - following an error, `current_size` is unchanged and the limit handling logic is applied as usual.